src/dirwatcher/dirwatcher.js

```diff
@@ -0,0 +1,52 @@
+import fs from 'fs';
+import EventEmitter from 'events';
+class MyEmitter extends EventEmitter{};
+
+export class DirWatcher{
+    watch(path, delay){
+        this.myEmitter = new MyEmitter();
+        let files, files2;
+        setInterval(()=>{
+            files2 = fs.readdirSync(path);
+            if(files){
+                // For addition of files
+                let change = this.compare(files, files2);
+                if(change) {
+                    this.myEmitter.emit('changed', change);
+                }
+                else{
+                    //For changes in any files.
+                    files.forEach((file, index) => {
+                        if((Date.now() - fs.statSync(path+'/'+file).mtimeMs) < 1000){
+                            this.myEmitter.emit('changed',files2[index]);
+                        }
+                    });
+                }
+            }
+            files = fs.readdirSync(path);
+        }, delay);
+    }
+
+    compare(arr1, arr2){
+        let el;
```

Comment:
Please 2 moments:
1. avoid undefined initialization of variables, always have default value to avoid type problems in future (there might be some cases to have undefined value, but very rarely)
2. PLease use more descriptive variable names, because arr1, arr2, el or other names are really poor and meaningless

Reply:
Updated. Thanks